### PR TITLE
WIP: Save also the original version of an image

### PR DESCRIPTION
**Pull request by juho-jaakkola**
Just something I wrote up quickly while doing manual testing for #9970

### CODE REVIEW

File: engine/classes/Elgg/EntityIconService.php [hunk 1]
```diff
@@ -247,14 +247,23 @@ public function saveIcon(ElggEntity $entity, ElggFile $file, $type = 'icon', arr
...
+
 		foreach ($sizes as $size => $opts) {
 
 			$width = (int) elgg_extract('w', $opts);
 			$height = (int) elgg_extract('h', $opts);
 			$square = (bool) elgg_extract('square', $opts);
 			$upscale = (bool) elgg_extract('upscale', $opts);
 
-			if ($cropping_mode && $square === false && $size !== 'large') {
+			if (type === 'icon' && $cropping_mode && $square === false && $size !== 'large') {
```
Comment (juho-jaakkola) [inline engine/classes/Elgg/EntityIconService.php]: This change will be added in #9971

**Comment by hypeJunction** [comment]
This only makes for non-file entities. Otherwise you end up with 2 copies of image file entity.

**Comment by hypeJunction** [comment]
Wouldn't it be easier to just skip master when in cropping mode?

**Comment by juho-jaakkola** [comment]
> This only makes for non-file entities. Otherwise you end up with 2 copies of image file entity.

I'm not sure I follow

> Wouldn't it be easier to just skip master when in cropping mode?

Master is also scaled down. I'd rather have one with genuinely good quality.

**Comment by hypeJunction** [comment]
Image file entities will end up with 2 copies of the same file

**Comment by juho-jaakkola** [comment]
Do you mean that when `saveIcon()` is called through `saveIconFromElggFile()` it makes a exact copy of a file that already exists?

**Comment by hypeJunction** [comment]
Yes. If you have file a entity that's an image, your code would create an exact copy of that image as an original icon.

**Pull request closed without merging**